src/txn/txn_rollback_to_stable.c

```diff
@@ -560,13 +569,19 @@ __rollback_abort_newer_row_leaf(
      */
     WT_ROW_FOREACH (page, rip, i) {
         if ((upd = WT_ROW_UPDATE(page, rip)) != NULL)
-            __rollback_abort_newer_update(session, upd, rollback_timestamp);
+            __rollback_abort_newer_update(session, upd, rollback_timestamp, &stable_update_found);
+        else
+            /*
+             * Reset the flag to make sure that we don't reuse an earlier row flag result for the
+             * current row.
+             */
+            stable_update_found = false;
 
         if ((insert = WT_ROW_INSERT(page, rip)) != NULL)
             __rollback_abort_newer_insert(session, insert, rollback_timestamp);
 
         /* If the configuration is not in-memory, abort any on-disk value. */
-        if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
+        if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY) && !stable_update_found)
             WT_RET(__rollback_abort_row_ondisk_kv(session, page, rip, rollback_timestamp));
```

Comment:
Why don't we move this to else part, where we set the stable_update_found to false ?

Reply:
No, we cannot move this into the else part of the code where `stble_upate_found` is set to false. The on-disk update needs to checked whenever there are no updates in the update list also.